ant-node-manager/src/lib.rs

```diff
@@ -558,12 +560,32 @@ pub async fn refresh_node_registry(
 ) -> Result<()> {
     // This message is useful for users, but needs to be suppressed when a JSON output is
     // requested.
+
     if print_refresh_message {
         println!("Refreshing the node registry...");
     }
     info!("Refreshing the node registry");
 
-    for node in &mut node_registry.nodes {
+    let total_nodes = node_registry.nodes.len();
+    let (tx, mut rx) = mpsc::channel::<usize>(100);
+
+    // Progress printer task
+    let handle = task::spawn(async move {
+        let mut last_percent = 0;
+        while let Some(processed) = rx.recv().await {
+            let percent = (processed * 100) / total_nodes.max(1);
+            if percent != last_percent {
+                use std::io::{stdout, Write};
+                stdout().flush().unwrap();
+                print!("\rRefresh progress - {:>3}%", percent);
```

Comment:
Can we use `indicatif::ProgressBar` here? It is already being used inside antnode manager.